### PR TITLE
Movies detail page

### LINKED ISSUE #8: Movie Details Page
As an authenticated user,
When I visit the movie's detail page,
I should see
- [x] Button to create a viewing party

**Details** This button should take the authenticated user to the new event page

And I should see the following information about the movie:

- [x] Movie Title
- [x] Vote Average of the movie
- [x] Runtime in hours & minutes
- [x] Genere(s) associated to movie
- [x] Summary description
- [x] List the first 10 cast members (characters&actress/actors)
- [x] Count of total reviews
- [x] Each review's author and information

**Details:** This information should come from 3 different endpoints from [The Movie DB API](https://developers.themoviedb.org/3/getting-started/introduction)

### PR DESCRIPTION
**Pull request by abreaux26**
__X__ Wrote Tests ____ Implemented ____ Reviewed

Neccesary checkmarks:

  - [x] All Tests are Passing
  - [x] The code will run locally
  - [x] Ran RuboCop `bundle exec rubocop --safe-auto-correct`

Type of change

  - [x] New feature
  - [ ] Bug Fix

Implements/Fixes:

    description closes #

Check the correct boxes

  - [x] This broke nothing
  - [ ] This broke some stuff
  - [ ] This broke everything

Testing Changes

  - [x] No Tests have been changed
  - [ ] Some Tests have been changed
  - [ ] All of the Tests have been changed(Please describe what in the world happened)

Checklist:

  - [ ] My code has no unused/commented out code
  - [x] I have reviewed my code
  - [ ] I have commented my code, particularly in hard-to-understand areas
  - [ ] I have fully tested my code

(For Fun!)Please Include a link to a gif of your feelings about this branch

Link:

### CODE REVIEW

File: app/controllers/movies_controller.rb [hunk 1]
```diff
@@ -8,4 +8,12 @@ def search
       @search = @movie_service.search(params[:q])
     end
   end
+
+  def detail
+    @user = current_user
+    @movie = @movie_service.movie_detail_for(params[:id])
+    @genres = @movie_service.genres(params[:id])
+    @casts = @movie_service.casts(params[:id])
+    @reviews = @movie_service.reviews(params[:id])
```
Comment (Trevorsuter) [inline app/controllers/movies_controller.rb:17]: What if we created a class for all of these objects? (Mainly for the purpose of DRYing out the controller

File: config/routes.rb [hunk 2]
```diff
@@ -10,5 +10,5 @@
   post '/dashboard', to: 'user_friends#create'
   get '/discover', to: 'movies#discover', as: 'discover'
   get '/movies', to: 'movies#search', as: 'movies'
-  get '/movies/:id', to: 'movies#details', as: 'movie'
+  get '/movies/:id', to: 'movies#detail', as: 'movie'
```
Comment (Trevorsuter) [inline config/routes.rb:13]: nice catch on this one.

File: spec/features/movies/detail_spec.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  it 'I see information about the movie' do
+    visit discover_path
+
+    VCR.use_cassette('movie_data_details') do
+      fill_in :q, with: 'Phoenix'
+      click_on('Search by Movie Title')
+
+      movie_title = find(".title", match: :first).text
+      vote_average = find(".vote_average", match: :first).text
```
Comment (Trevorsuter) [inline spec/features/movies/detail_spec.rb:16]: This is cool.. Is this capybara?

**Review by Trevorsuter — APPROVED** [review]
Nice work Angel!!

File: spec/features/movies/detail_spec.rb (hunk 3 above)
Comment (abreaux26) [reply spec/features/movies/detail_spec.rb:16]: Yes! Lots of research 😅 and me just not thinking lol

File: app/controllers/movies_controller.rb (hunk 1 above)
Comment (abreaux26) [reply app/controllers/movies_controller.rb:17]: yess!! I think we can do this when we refactor!

**Pull request merged**